plugins/code-climate/src/plugin.ts

```diff
@@ -39,7 +39,8 @@ export const codeClimatePlugin = createPlugin({
         discoveryApi: discoveryApiRef,
         identityApi: identityApiRef,
       },
-      factory: ({ discoveryApi }) => new ProductionCodeClimateApi(discoveryApi),
+      factory: ({ discoveryApi, identityApi }) =>
```

Comment:
You could pass in the [fetchApi](https://github.com/backstage/backstage/blob/master/packages/core-plugin-api/src/apis/definitions/FetchApi.ts) here instead and the token would be handled for you :)

Reply:
Thanks @jhaals. Wasn't aware of the fetchApi. I have updated changes in [9a20067](https://github.com/backstage/backstage/pull/13103/commits/9a20067f441d142e5c04412742e494a573280ffd)